### PR TITLE
LSP: Format Document and Format Selection actions implemented.

### PR DESCRIPTION
**Pull request by dbalek**
LSP: `Format Document` and `Format Selection` actions implemented.
The implementation can read formatting settings exported form NetBeans.

### CODE REVIEW

File: groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/lexer/LexUtilities.java [hunk 1]
```diff
@@ -446,6 +511,15 @@ public static boolean isBeginToken(TokenId id, BaseDocument doc, int offset) {
         return END_PAIRS.contains(id);
     }
 
+    /**
+     * Return true iff the given token is a token that should be matched
```
Comment (ppisl) [inline groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/lexer/LexUtilities.java:515]: Possible typo here 'iff' . If it's really typo, it appears in more places.

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java [hunk 2]
```diff
@@ -2243,4 +2285,220 @@ protected CallHierarchyOutgoingCall createResultItem(CallHierarchyItem item, Lis
         };
         return t.processRequest();
     }
+
```
Comment (ppisl) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java:2288]: This is not against this PR, but this file is really long :(

**Review by ppisl — APPROVED** [review]
From my point of view, OK.

File: groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/lexer/LexUtilities.java (hunk 1 above)
Comment (dbalek) [reply groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/lexer/LexUtilities.java:515]: 'iff' usually means 'if and only if'.

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java (hunk 2 above)
Comment (dbalek) [reply java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java:2288]: Agree :-(

**Pull request merged**